web/src/components/MoreOptionsDropdown.tsx

```diff
@@ -0,0 +1,215 @@
+import * as DropdownMenu from '@radix-ui/react-dropdown-menu';
+import { useFeatureFlag } from 'features/feature-flags/api';
+import { useShare } from 'hooks/useShare';
+import { useAtomValue } from 'jotai';
+import { Link, ShareIcon } from 'lucide-react';
+import { useMemo, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { FaFacebook, FaLinkedin, FaReddit, FaSquareXTwitter } from 'react-icons/fa6';
+import { twMerge } from 'tailwind-merge';
+import { ShareType, trackShareChart } from 'utils/analytics';
+import { baseUrl, DEFAULT_ICON_SIZE, DEFAULT_TOAST_DURATION } from 'utils/constants';
+import { hasMobileUserAgent as hasMobileUA } from 'utils/helpers';
+import { displayByEmissionsAtom, isHourlyAtom } from 'utils/state/atoms';
+
+import { DefaultCloseButton } from './DefaultCloseButton';
+import { TimeDisplay } from './TimeDisplay';
+import { Toast, useToastReference } from './Toast';
+
+// TODO: add chartId to tracking
+const onTrackShareChartReddit = trackShareChart(ShareType.REDDIT);
+const onTrackShareChartTwitter = trackShareChart(ShareType.TWITTER);
+const onTrackShareChartLinkedin = trackShareChart(ShareType.LINKEDIN);
+const onTrackShareChartFacebook = trackShareChart(ShareType.FACEBOOK);
+const onTrackShareChart = trackShareChart(ShareType.SHARE);
+
+export interface MoreOptionsDropdownProps {
+  children: React.ReactElement;
+  shareUrl?: string;
```

Comment:
Currently baseUrl; will make a follow up PR once this one and https://github.com/electricitymaps/electricitymaps-contrib/pull/7333 gets merged in that will create the chart shareUrl